adam-core/src/main/scala/org/bdgenomics/adam/rdd/features/CoverageRDD.scala

```diff
@@ -0,0 +1,85 @@
+/**
+ * Licensed to Big Data Genomics (BDG) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The BDG licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bdgenomics.adam.rdd.features
+
+import org.apache.spark.rdd.RDD
+import org.bdgenomics.adam.models.ReferenceRegion
+
+/**
+ * RDD holding run length encoded coverage in a FeatureRDD. CoverageRDD stores
+ * coverage counts at a ReferenceRegion in the score attribute of
+ * a feature.
+ *
+ * @param featureRDD FeatureRDD holding records for coverage and corresponding
+ * referenceRegion
+ */
+case class CoverageRDD(featureRDD: FeatureRDD) {
+
+  /**
+   * Java friendly save function. Automatically detects the output format.
+   *
+   * If the filename ends in ".bed", we write a BED file. If the file name ends
+   * in ".narrow[pP]eak", we save in the NarrowPeak format. If the file name
+   * ends in ".interval_list", we save in the interval list format. Else, we
+   * save as Parquet. These files are written as sharded text files.
+   *
+   * @param filePath The location to write the output.
+   */
+  def save(filePath: java.lang.String) = {
+    featureRDD.save(filePath)
+  }
+
+  /**
+   * Gets coverage overlapping specified ReferenceRegion. For large, ReferenceRegions,
+   * base pairs per bin (bpPerBin) can be specified to bin together ReferenceRegions of
+   * equal size.
+   *
+   * @param region ReferenceRegion to fetch overlapping coverage from
+   * @param bpPerBin base pairs per bin, number of bases to combine to one bin
```

Comment:
This function does a mod for bpPerBin instead of aggregating the bin. However, I couldn't think of how to computing the average size of coverage bins without shuffling. Is this ok?